src/main/java/com/silzila/querybuilder/RelativeCalendarFilter.java

```diff
@@ -0,0 +1,70 @@
+package com.silzila.querybuilder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.h2.expression.Alias;
+
+import com.silzila.helper.AilasMaker;
+import com.silzila.payload.request.Dimension;
+import com.silzila.payload.request.Measure;
+import com.silzila.payload.request.Query;
+
+public class RelativeCalendarFilter {
+
+    public static String buildRelativeCalendarFilter(Query req, String CTEquery, String relativeType) {
+
+        Map<String, Integer> aliasNumbering = new HashMap<>();
+
+        String CTEqueryString = "WITH CTE AS (\n" + CTEquery + "\n) \n";
+
+        List<Dimension> dimensions = req.getDimensions();
+
+        List<Measure> measures = req.getMeasures();
+
+        List<String> selectWindowClause = new ArrayList<>();
+
+        for (Measure measure : measures) {
+
+            String alias = AilasMaker.aliasing(measure.getFieldName(), aliasNumbering);
+            String field = measure.getAggr().name() + "(" + alias
+                    + ") over (" + partitionBy(dimensions) + ")";
+            selectWindowClause.add(field);
+        }
+
+        StringBuilder finalRelativeQuery = new StringBuilder();
+        finalRelativeQuery.append(CTEqueryString);
+        finalRelativeQuery.append("Select * , ");
+        finalRelativeQuery.append(String.join(", ", selectWindowClause));
```

Comment:
if there isn't any value in selectWindowClause, then your query will look something like this
`Select * , from CTE`. Will this work?

Reply:
remove this section if it is not used!